Lib9c/Action/Buy0.cs

```diff
@@ -60,7 +61,9 @@ public override IAccountStateDelta Execute(IActionContext context)
                         GoldCurrencyState.Address);
                 return states.SetState(ShopState.Address, MarkChanged);
             }
-            
+
+            CheckObsolete(1815592, context);
```

Comment:
It isn't matched with [`ActionObsolete()`'s](https://github.com/planetarium/lib9c/pull/487/files#diff-03b0cf6730ee3e7c83a692d427111da44c44a6c0ea29d4a71211793a14d1d77cR20). is there any reason?

Reply:
This matches the CheckPermission in the IActionContext extension. Shall we change the name?